Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.Ballistic.cs

```diff
@@ -71,18 +77,63 @@ private void OnBallisticAfterInteract(EntityUid uid, BallisticAmmoProviderCompon
             Deleted(args.Target) ||
             !TryComp<BallisticAmmoProviderComponent>(args.Target, out var targetComponent) ||
             targetComponent.Whitelist == null)
-        {
             return;
-        }
 
         args.Handled = true;
 
-        // Continuous loading
-        _doAfter.TryStartDoAfter(new DoAfterArgs(EntityManager, args.User, component.FillDelay, new AmmoFillDoAfterEvent(), used: uid, target: args.Target, eventTarget: uid)
+        ReloadDoAfter(component, args.User, args.Target.Value, args.Used);
+    }
+
+    private void OnBallisticAfterInteractUsing(EntityUid uid, BallisticAmmoProviderComponent component, AfterInteractUsingEvent args)
```

Comment:
You aren't checking for if the item being interacted with is one that can actually load shells here, which means the doafter fires when interacting with seemingly any object. 

Try clicking on the bandolier with a hamburger while wearing it and you'll see what I mean.